### ISSUE TITLE
Outlier tests

### ISSUE THREAD

**Issue by Beakerboy** [markrogoyski/math-php#273]
It would be useful to have outlier tests like Grubbs and Tietjen-Moore. They both work on univariate data. How would you prefer a function like these work? If an array of n values is fed into the function, would you like it to return an array without the outliers? Alternatively, two function can be made for each, one to calculate the critical values, and one for the test statistic. The user can then choose what to do with the information.

**Comment by markrogoyski** [comment]
This sounds like a good feature to add.

Are you thinking something like this?
```php
$dataSet = [1, 1, 1, 2, 2, 2, 685874, 2, 2, 1, 1];

$tmTestStatistic  = Statistics\Outliers\TietjenMore::testStatistic($dataSet);
$outliers         = Statistics\Outliers\TietjenMore::outliers($dataSet);
$numberOfOutliers = Statistics\Outliers\TietjenMore::numberOfOutliers($dataSet);
```

Are there other ways to provide this interface or to model it that makes it simple to understand and easy to use for the end user?

**Comment by Beakerboy** [comment]
Neither of these two tests can give you the number of outliers. Grubbs only tests the most extreme value, while Tietjen-Moore requires the analyst to provide a number. I don't know if TM requires that all be on the same side of the mean, or if it just looks at the top k residuals. I'm about to give a patch for the Grubbs with the standard test data for this procedure defined by NIST. Look it over and see where you would like to put it, and I can do any subsequent outlier detection algorithms similarly.

**Comment by Beakerboy** [comment]
Pull Request:
https://github.com/markrogoyski/math-php/pull/274

**Comment by markrogoyski** [comment]
Grubbs' test has been released in [v0.52.0](https://github.com/markrogoyski/math-php/releases/tag/v0.52.0). Thanks for suggesting this and working on the initial version.

Let's try to add some more outlier tests after V1.0.0 is released and perhaps come up with a standard interface to remove outliers from data sets in v2.0.0.
